notif.go

```diff
@@ -33,34 +29,54 @@ func (nn *netNotifiee) Connected(n inet.Network, v inet.Conn) {
 	default:
 	}
 
+	dht.plk.Lock()
+	defer dht.plk.Unlock()
```

Comment:
Let's maybe do an explicit unlock here as there is quite large `go fun()` body lower.
It is just my feeling while reading the code I had to scroll down all the way to check what happens next and if the lock isn't held too long.